Content for Git Pull Request Page (#78)
axelerant/engg-handbook: content/docs/how-we-work/git/pull-request.md

---
title: Pull Request (PR)
weight: 3
description: A Git PR, or pull request, is a way to propose changes to the codebase. Post review your changes are merged. This process helps to ensure that changes are reviewed by someone else, which reduces the chances of any errors.
---

# Pull Request (PR)

A Git PR, or pull request, is a way to propose changes to the codebase. Post review your changes are merged. This process helps to ensure that changes are reviewed by someone else, which reduces the chances of any errors.

> By utilizing Git commands, we can [create and submit a pull request](https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project#:~:text=Creating%20a%20Pull%20Request) to the repository owner and manage the entire process of review, approval, and merging of changes. The process involves creating a fork of the repository, cloning the fork to a local computer, making desired changes to the code, and committing these changes back to the repository. By submitting a PR, users can demonstrate how they have added value or addressed issues in the codebase, show evidence of their contributions, and allow the repository owner to evaluate the proposed changes prior to incorporating them into the codebase.

Reviewer: Git commands are only useful to commit and push changes. The actual PR change happens via a UI of some sort or using a tool like `gh` which is out of scope here.

```suggestion
We can [create and submit a pull request](https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project#:~:text=Creating%20a%20Pull%20Request) to the repository owner and manage the entire process of review, approval, and merging of changes. The process involves creating a fork of the repository, cloning the fork to a local computer, making desired changes to the code, and committing these changes. A PR can be submitted once these changes are pushed. By submitting a PR, users allow the repository owner to evaluate the proposed changes prior to incorporating them into the codebase.
```
